File: sqlbuilder/sqlbuilder/Disjunction.java
package com.clickability.cms.dataaccess.sqlbuilder;

import java.util.ArrayList;
import java.util.List;

public class Disjunction implements Expression {

	private List<Expression> predicates;
	private Expression first;

	public Disjunction(Expression first) {
		this.predicates = new ArrayList<Expression>();
		this.first = first;
	}

	public Disjunction or(Expression... clauses) {
		for (Expression c: clauses) {
			this.predicates.add(c);
		} 
		return this;
	}
	
	public Integer size() {
		return this.predicates.size() + 1;
	}
	
	@Override
	public String get() {
		String result = first.get();
		for (Expression predicate: this.predicates) {
			result += String.format(" or (%s)", predicate.get());
		}
		return result;
	}

}

File: java/queryBuilder/Disjunction.kt
package com.clickability.cms.dataaccess.sqlbuilder

import java.util.ArrayList

class Disjunction(private val first: Expression) : Expression {

    private val predicates: List<Expression>

    init {
        this.predicates = ArrayList<Expression>()
    }

    fun or(vararg clauses: Expression): Disjunction {
        for (c in clauses) {
            this.predicates.add(c)
        }
        return this
    }

    fun size(): Integer {
        return this.predicates.size() + 1
    }

    @Override
    fun get(): String {
        var result = first.get()
        for (predicate in this.predicates) {
            result += String.format(" or (%s)", predicate.get())
        }
        return result
    }

}
